### PR TITLE
Catch CrashError of smoldot, restart smoldot and add UI indications

### PR DESCRIPTION
**Pull request by wirednkod**
This PR takes care of the possible smoldot panic/error that crashes the extension - and addresses the last bullet on [Pierre's comment](https://github.com/paritytech/substrate-connect/issues/429#issuecomment-1059754226) and issue [#507](https://github.com/paritytech/substrate-connect/issues/507)
When an error happens on smoldot client, the extension catches the error, prints it in the logs <del>and starts a series of extension restarts. There will be 3 restarts of the extension with a 5 seconds delay between each restart.</del>
If <del>at the 3rd retry</del> a CrashError occurs <del>again</del>, then 2 icons appear on the Popup window. 
1 "alert" icon that indicates the issue in the extension and user can hover and read the error. If icon is clicked user is redirected to "paused" logs (that contain ofc the error log last - that lead to the crash)

I did not follow exactly the proposal of Pierre on the 4th bullet of his comment - meaning that if smoldot crashes but recovers the only thing that will happen will be to get disconnected from apps and try to run smoldot again. 

The only way that the _average Joe_ will understand that something really happened is if smoldot crashes 3 consecutive times and then the indication will appear (and the manual "restart extension" button)

EDIT: This covers a potential crash only during initialization of smoldot

image
image

### CODE REVIEW

**Comment by tomaka** [comment]
This handles the situation where smoldot crashes at initialization, but this is extremely rare.
Handling the crashes while it's running is way more difficult and I've laid down some ground but haven't finished.

**Comment by wirednkod** [comment]
> This handles the situation where smoldot crashes at initialization, but this is extremely rare.
> Handling the crashes while it's running is way more difficult and I've laid down some ground but haven't finished.
> 

That is correct. This covers the crash when smoldot inits. We had plenty of that in past weeks and it would be good to catch them. Will remove the retries and allow just the manual retry

File: projects/extension/src/containers/Popup.tsx [hunk 1]
```diff
@@ -1,23 +1,46 @@
...
+  const [crashError, setCrashError] = useState<string>("")
   const appliedTheme = createTheme(light)
 
   useEffect(() => {
     let isActive = true
     let unsubscribe = () => {}
 
+    chrome.storage.local.get(["crashError"], (res) => {
+      setCrashError(res.crashError)
+    })
```
Comment (tomaka) [inline projects/extension/src/containers/Popup.tsx:43]: Does this auto-update if you call `chrome.storage.local.set(["crashError"], ...)`? I guess no?

File: projects/extension/src/background/index.ts [hunk 2]
```diff
@@ -35,6 +35,12 @@ export interface Background extends Window {
   }
 }
 
+interface ErrorStruct {
+  name: string
+  message: string
+  stack?: string
+}
```
Comment (tomaka) [inline projects/extension/src/background/index.ts:42]: Why is this necessary? Isn't this the `Error` interface?

File: projects/extension/src/containers/Options.tsx [hunk 3]
```diff
@@ -184,6 +184,12 @@ const Options: React.FunctionComponent = () => {
   }, [poolingLogs])
 
   useEffect(() => {
+    chrome.storage.local.get(["crashError"], (res) => {
+      if (res.crashError) {
+        setValue(2)
```
Comment (tomaka) [inline projects/extension/src/containers/Options.tsx:189]: What's "value"? Can we have a better name, or if not at the very least a comment?

File: projects/extension/src/containers/Popup.tsx (hunk 1 above)
Comment (wirednkod) [reply projects/extension/src/containers/Popup.tsx:43]: This updates every time the popup opens.If popup is open and the localStorage update then no - but based on the nature of the popup thats really an edge scenario

**Comment by wirednkod** [comment]
Due to the major changes that #984 brings, this will close and will re-implemented with Goran's designs
